apperrors/apperrors.go

```diff
@@ -0,0 +1,69 @@
+package apperrors
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	l "github.com/sirupsen/logrus"
+)
+
+// ErrorStruct - struct used to convert error messages into required JSON format
+type ErrorStruct struct {
+	Message string `json:"message,omitempty"` //Error Message
+	Status  int    `json:"status,omitempty"`  //HTTP Response status code
+}
+
+// Error - prints out an error
+func Error(appError error, msg string, triggeringError error) {
+	l.WithFields(l.Fields{"appError": appError, "message": msg}).Error(triggeringError)
+}
+
+// Warn - for warnings
+func Warn(appError error, msg string, triggeringError error) {
+	l.WithFields(l.Fields{"appError": appError, "message": msg}).Warn(triggeringError)
+}
+
+// JSONError - This function writes out an error response with the status
+// header passed in
+func JSONError(rw http.ResponseWriter, status int, err error) {
```

Comment:
`rw http.ResponseWriter` should not be part of `apperros` package.

instead create new package call `apphttp` & put all code in

`WriteJSONError(rw http.ResponseWriter, status int, err error)`